SendFriendRequestTests

Zargess opened the pull request:
Removed unused field in MasterLogin
Tested the flow of sending af FriendRequest from the client

Commit:
SendFriendRequestTests

Removed unused field in MasterLogin
Tested the flow of sending af FriendRequest from the client

miniwolf commented on fagiClient/src/com/fagi/view/content/InviteItem.fxml:
```diff
@@ -2,12 +2,9 @@
-<?import javafx.scene.layout.HBox?>
-<?import javafx.scene.layout.Pane?>
-<?import javafx.scene.layout.VBox?>
+<?import javafx.scene.layout.*?>
```
You need to change your settings to allow single class import.

miniwolf commented on fagiClient/src/com/fagi/view/conversation/Invitation.fxml:
```diff
@@ -1,16 +1,10 @@
-<?import javafx.scene.layout.VBox?>
-
-<fx:root type="BorderPane" styleClass="message" xmlns="http://javafx.com/javafx/8.0.60"
+<?import javafx.scene.control.*?>
```
You need to change your settings to allow single class import.

miniwolf replied:
This is still a problem here.

miniwolf commented on test/src/com/fagi/guitests/SearchUserTests.java:
```diff
@@ -164,9 +164,10 @@ public void WhenSearchingForUsers_TheResultingNamesMustBeVisible(FxRobot robot)
     public void WhenSearchingForUsers_TheResultingProfilePicturesShouldBeVisible(FxRobot robot) {
         var usernames = new ArrayList<String>() {{ add("a"); add("ab"); }};
-        addIngoingMessageToInputHandler(new SearchUsersResult(usernames, new ArrayList<>()));
+        addIngoingMessageToInputHandler(inputHandler, new SearchUsersResult(usernames, new ArrayList<>()));
```
Please go over all added calls, ensuring that we are using the appropriate amount of retries.

miniwolf replied:
Also make sure that we have no more inputHandler.AddInGoingMessage laying around. Like in FriendsTests.java for instance.

miniwolf commented on test/src/com/fagi/guitests/SendFriendRequestTests.java:
```diff
@@ -0,0 +1,136 @@
+    void WhenOpeningANewFriendInvitation_SendInvitationContainsContactUserName(FxRobot robot) {
+        var username = "a";
+        var usernames = new ArrayList<String>() {{ add(username); }};
+        addIngoingMessageToInputHandler(inputHandler, new SearchUsersResult(usernames, new ArrayList<>()), 2);
```
Why 2 here?

miniwolf commented on test/src/com/fagi/guitests/SendFriendRequestTests.java:
```diff
@@ -0,0 +1,136 @@
+        addIngoingMessageToInputHandler(inputHandler, new SearchUsersResult(usernames, new ArrayList<>()), 2);
+        Assume.assumeNotNull(robot.clickOn("#UniqueSearchContact"));
+        Assume.assumeTrue(robot.lookup("#InvitationConversation").tryQuery().isPresent());
+        Label nameLabel = (Label) robot.lookup("#name").tryQuery().orElse(null);
```
You need to test assert this or assume this.

miniwolf commented on test/src/com/fagi/guitests/SendFriendRequestTests.java:
```diff
@@ -0,0 +1,136 @@
+    void WhenClickingOnSendInvitationButton_FriendRequestIsSendToServer(FxRobot robot) {
+        var username = "a";
+        var usernames = new ArrayList<String>() {{ add(username); }};
+        addIngoingMessageToInputHandler(inputHandler, new SearchUsersResult(usernames, new ArrayList<>()), 2);
```
why 2 here?

miniwolf commented on test/src/com/fagi/guitests/SendFriendRequestTests.java:
```diff
@@ -0,0 +1,136 @@
+    void WhenClickingOnSendInvitationButton_TheWrittenMessageIsIncluded(FxRobot robot) {var username = "a";
+        var usernames = new ArrayList<String>() {{ add(username); }};
+        var message = "Do you want to build a snowman?";
+        addIngoingMessageToInputHandler(inputHandler, new SearchUsersResult(usernames, new ArrayList<>()), 2);
```
why 2 here?

miniwolf commented on test/src/com/fagi/guitests/SendFriendRequestTests.java:
```diff
@@ -0,0 +1,136 @@
+        addIngoingMessageToInputHandler(inputHandler, new SearchUsersResult(usernames, new ArrayList<>()), 2);
+        Assume.assumeNotNull(robot.clickOn("#UniqueSearchContact"));
+        Assume.assumeTrue(robot.lookup("#InvitationConversation").tryQuery().isPresent());
+        Assert.assertTrue(robot.lookup("#send").tryQuery().isPresent());
```
Having 2 asserts should be split in two tests. This is only the case where you have assert followed by new code execution followed by new assert.

miniwolf commented on test/src/com/fagi/guitests/SendFriendRequestTests.java:
```diff
@@ -0,0 +1,136 @@
+        Assume.assumeTrue(robot.lookup("#InvitationConversation").tryQuery().isPresent());
+
+        Node messageField = robot.lookup("#message").tryQuery().orElse(null);
+        Assert.assertNotNull(messageField);
```
Having 2 asserts should be split in two tests.

Commit:
SendFriendRequestTests

Imports single classes again
Made tests use new helper functions for inserting messages into inputhandler

Commit:
SendFriendRequestTests

Found library for easy work with optionals when asserting.

miniwolf committed:
Add assertj to test runners.
